### PR TITLE
Editor: Fix error when creating new record

### PR DESCRIPTION
**Pull request by tkohr**
### Description

The goal of this PR is to fix the error when creating a new record. The same exception sometimes occurs when editing fields of a draft. A first commit ensures errors displayed via the `notifications.service` still to be logged to the console.

The actual error is caused by the fact that the `EditRecordResolver` is called even when routing to `/create`. This happens due to the `navigate['edit']` calls within the component. Both navigations have been removed:
- 07dcd58b154bf47f0bf0d5e76f001e703123cbef does not seem essential, but maybe I've missed something. Creating a new record now stays on `/create`. => This prevents the exception when creating a new record.
- a97b566ab19a4a98b0ef3831aa8da821642a4a49 does not seem to be used anymore, since we've replaced the `uniqueIdentifier` with the `resourceIdentifier ` in the form. => This prevents the exception when editing fields of a draft.

Note to reproduce: The issue occurs on the first access to `/create` after app reload. It starts with the second access to `/create` or accessing `/create` after `/edit`. 

### Screenshots

image

### Quality Assurance Checklist

- [ ] Commit history is devoid of any _merge commits_ and readable to facilitate reviews
- [x] If **new logic** ⚙️ is introduced: unit tests were added
- [ ] If **new user stories** 🤏 are introduced: E2E tests were added
- [ ] If **new UI components** 🕹️ are introduced: corresponding stories in Storybook were created
- [ ] If **breaking changes** 🪚 are introduced: add the `breaking change` label
- [ ] If **bugs** 🐞 are fixed: add the `backport <release branch>` label
- [ ] The [documentation website](docs) 📚 has received the love it deserves

---

### CODE REVIEW

**Comment by jahow** [comment]
I'm not sure that this fix is the right way forward; the automatic redirection from /create to /edit/TEMP-ID-XXX was made so that refreshing the page was not going to create a new record but instead leave the user on the same draft.

Maybe if it's just a timing issue a different fix could be found? I'm taking a look right now

File: apps/metadata-editor/src/app/edit/edit-page.component.spec.ts
Comment (jahow) [inline apps/metadata-editor/src/app/edit/edit-page.component.spec.ts:174]: This is actually needed because when a record is first saved its ID is replaced from a temporary one (TEMP-ID-xxxx) to a final UUID

File: apps/metadata-editor/src/app/edit/edit-page.component.ts
Comment (jahow) [inline apps/metadata-editor/src/app/edit/edit-page.component.ts:153]: This is actually needed because when a record is first saved its ID is replaced from a temporary one (TEMP-ID-xxxx) to a final UUID

**Comment by tkohr** [comment]
Thanks for the feedback @jahow! I didn't think of reloading the app on create, good catch!

I've noticed that the reverted `subscribe` that is not yet added to the component's subscriptions still causes the error when editing draft records, so I also wrapped it into a `subscription.add()`. Also added some UTs for the subscriptions. I'll let you have a final look before merging.

File: libs/feature/editor/src/lib/components/import-record/import-record.component.spec.ts
Comment (jahow) [inline libs/feature/editor/src/lib/components/import-record/import-record.component.spec.ts:10]: to remove

File: apps/metadata-editor/src/app/edit/edit-page.component.ts
Comment (jahow) [inline apps/metadata-editor/src/app/edit/edit-page.component.ts:151]: This isn't needed because it's a short-live observable (there is a `take(1)` in the pipe); maybe getting rid of the subscription.add() and mentioning this fact in a comment will avoid confusion in the future?

**Review by jahow — APPROVED** [review]
Thank you! I just saw two new things that should be addressed, sorry :grimacing: I'll approve the PR and let you merge when you think it's done :)

File: libs/feature/editor/src/lib/components/import-record/import-record.component.spec.ts
Comment (tkohr) [reply libs/feature/editor/src/lib/components/import-record/import-record.component.spec.ts]: thanks, I've removed it

File: apps/metadata-editor/src/app/edit/edit-page.component.ts
Comment (tkohr) [reply apps/metadata-editor/src/app/edit/edit-page.component.ts:151]: As discussed, I've kept this due to the error when editing a draft record.

**Pull request merged**
